### PR TITLE
Notify user if they are using aemm commands on a non-aemm project

### LINKED ISSUE #21: Notify the user if they are using aemm commands on a non-aemm project
Currently, a user can run aemm commands on a cordova (non-aemm) project and get away with it with one notable exception: if the iOS project is named something other than CordovaPlugins.framework, then the ```package``` command will fail with:
```Error: No built framework. Please run `aemm build ios`, see `aemm help build` for details.```
It would be nice to let them know before they get to the ```package``` command, so they don't have to start over from ```project create```.

### PR DESCRIPTION
**Pull request by adzellman**
- Fixes #21
 - Add promises to many functions because the check for isAEMM parses XML
   and is asynchronous.
 - Add a no-samples flag (actually a samples flag, but in nopt it's the same thing)
   to generate a project without sample files in the www directory.
 - Clean up some unit tests because of a previous change to the argv object.
 - Ignore another negative test.

@HenryPing 👍  pending QE
@EonerAemm 👍  Did positive workflow on iOS, please merge once verifying Android workflow
@rissmanadobe 👍

### CODE REVIEW

File: src/android-emulator.js [hunk 1]
```diff
@@ -26,73 +26,75 @@ var getUserHome = require('../utils/getUserHome');
...
+            spinner.start();
+            checkBooted(port);
 
-        // start polling for device ready
-        spinner.start();
-        checkBooted(port);
-
-        return defer.promise;
+            return defer.promise;
+        });
```
Comment (HenryPing) [inline src/android-emulator.js]: Hard to follow from the diff, are the changes in this file simply re-alignment?

Comment (adzellman) [inline src/android-emulator.js]: `getValueFromConfig` is now asynchronous, so there is a slight change in the chaining besides the re-alignment

File: src/project.js [hunk 2]
```diff
@@ -136,7 +147,25 @@ function createAEMMScaffolding(fullProjectPath)
...
+			if (result.widget.name[0] === "CordovaPlugins") {
+				deferred.resolve(true);
+			}
+			else {
+				deferred.resolve(false);
+			}
+		});
+		return deferred.promise;
+	});
+}
```
Comment (HenryPing) [inline src/project.js]: Can we persist this info? to minimize the parse of xml which can be expensive. isAEMMProject() will be called a lot because of projectRootPath() calls.

Comment (adzellman) [inline src/project.js]: I assume you mean persist to a json file? Do you have a proposal for where that would live? There's no precedent for a persistent state object on the project.

Comment (HenryPing) [inline src/project.js]: Maybe create a file(i.e. aemm_version) at the project root specific to aemm during the creation of the project. Checking the existence of this file probably suffice.

Comment (adzellman) [inline src/project.js]: The error is actually a graceful-fs warning that is being tracked by #37

**Comment by HenryPing** [comment]
The change looks good to me, however, the scope of changes require full QE pass for both iOS and Android, on both MacOSX and Windows(Android only). I will be on PTO next week. @EonerAemm can merge it after it passes QE.
